mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -5662,4 +5666,85 @@ protected override StringBuilder AppendNamespace (StringBuilder buf, TypeReferen
 	}
 }
 
+internal abstract class FlagsFormatter {
+	public abstract string Format(long c, Dictionary<long, string> values, string typename);
```

Comment:
The advantage to the above approach is that we don't "bake in" underlying enum types into the API, which makes things saner if/when we allow external add-ins -- we want as few assumptions/dependencies as possible. Here, all we rely on is the ability to get a Cecil TypeDefinition, as opposed to "all enum values are able to be stored in `long`s".